pkg/app/api/grpcapi/web_api.go

```diff
@@ -168,6 +170,107 @@ func (a *WebAPI) ListEnvironments(ctx context.Context, req *webservice.ListEnvir
 	}, nil
 }
 
+func (a *WebAPI) EnableEnvironment(ctx context.Context, req *webservice.EnableEnvironmentRequest) (*webservice.EnableEnvironmentResponse, error) {
+	if err := a.updateEnvironmentEnable(ctx, req.EnvironmentId, true); err != nil {
+		return nil, err
+	}
+	return &webservice.EnableEnvironmentResponse{}, nil
+}
+
+func (a *WebAPI) DisableEnvironment(ctx context.Context, req *webservice.DisableEnvironmentRequest) (*webservice.DisableEnvironmentResponse, error) {
+	if err := a.updateEnvironmentEnable(ctx, req.EnvironmentId, false); err != nil {
+		return nil, err
+	}
+	return &webservice.DisableEnvironmentResponse{}, nil
+}
+
+func (a *WebAPI) DeleteEnvironment(ctx context.Context, req *webservice.DeleteEnvironmentRequest) (*webservice.DeleteEnvironmentResponse, error) {
```

Comment:
Didn't we decide that it should be checked to ensure no piped is using the env before allowing disabling/deleting it?
I'm not sure if I remember correctly.

Reply:
We did. I'm looking to discuss that on https://github.com/pipe-cd/pipe/issues/2017 as mentioned in the PR body.

Reply:
Nice. Let's go.